controllers/front/landing.php

```diff
@@ -83,20 +87,23 @@ private function renderProductList()
             $this->context->getTranslator()
         );
 
-        $products = [];
-        foreach ($this->products as $productDetail) {
-            $products[] = $presenter->present(
-                $presentationSettings,
-                $assembler->assembleProduct($productDetail),
-                $this->context->language
-            );
+        $products_length = count($this->products);
+        for ($i = 0; $i < $products_length; ++$i) {
```

Comment:
Por que no haces un foreach de $this->products?
```suggestion
        foreach ($this->products as $p) {
```

Reply:
Porque necesito la variable $i para la siguiente línea 100, aunque es verdad que podría crear un contador aparte. Lo hablamos por Slack :smile:

Reply:
Cambiado en: https://github.com/doofinder/doofinder-prestashop/pull/148/commits/867a73e408533a2a11af27d205ec6b208b86ffb7